### PR TITLE
Implement ability 'Lackey', 'Invoke' and 11 DRAGONS cards

### PR DESCRIPTION
**Pull request by utilForever**
This revision includes:
- Implement abilities
  - Lackey: Lackeys are uncollectible minion cards generated by certain cards in the Year of the Dragon. They are mostly restricted to Priest, Rogue, Shaman, Warlock, and Warrior.
  - Invoke: Invoke is an ability introduced in the Descent of Dragons expansion. When an Invoke card is played, the Hero Power of Galakrond will be activated, as long as the player has a Galakrond hero card in their hand, deck, or battlefield.
- Implement 11 DRAGONS cards
  - Corrosive Breath (DRG_006)
  - Stormhammer (DRG_007)
  - Righteous Cause (DRG_008)
  - Diving Gryphon (DRG_010)
  - Gyrocopter (DRG_061)
  - Disciple of Galakrond (DRG_303)
  - Galakrond, the Wretched (DRG_600)
  - Galakrond, the Nightmare (DRG_610)
  - Galakrond, the Tempest (DRG_620)
  - Galakrond, the Unbreakable (DRG_650)
  - Galakrond, the Unspeakable (DRG_660)
- Implement tasks
  - AddLackeyTask: This class represents the task for adding Lackey to hand.
  - InvokeTask: This class represents the task for processing ability 'Invoke'.
- Add functions and methods
  - PlayHero(): Plays a hero card from player's hand.
  - IsGalakrond(): Returns the flag that indicates whether it is Galakrond.
  - GetLackeys(): Returns a list of Lackey cards.
  - EndsWith(): Finds out if value ends with ending.
  - IsHoldingRace(): SelfCondition wrapper for checking the player has entity with race in hand zone.
  - IsRush(): SelfCondition wrapper for checking the entity is rush.
  - IsLackey(): Returns the flag that indicates whether it is Lackey.
  - SetHero(): Sets the hero of the player.
  - UpgradeGalakrond(): Upgrades the Galakrond hero card.
  - GetInvoke(): Returns the value of invoke.
  - IncreaseInvoke(): Increases the value of invoke.
  - IsImmune(): Returns the flag that indicates whether it is immune.
- Fix `Hero::GetAttack()` mechanism (#304)

### CODE REVIEW

**Comment by utilForever** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- Tests/UnitTests/CardSets/DragonsCardsGenTests.cpp  474
- Sources/Rosetta/CardSets/DragonsCardsGen.cpp  9
- Tests/UnitTests/CardSets/DalaranCardsGenTests.cpp  20
         

Clones removed
==============
+ Sources/Rosetta/CardSets/UldumCardsGen.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/utilForever/RosettaStone/pullRequest?prid=5095569&bid=16760792)

File: Sources/Rosetta/Tasks/SimpleTasks/InvokeTask.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        return TaskStatus::COMPLETE;
+    }
+
+    // Upgrade Galakrond's Battlecry
+    player->IncreaseInvoke();
+
+    if (player->GetInvoke() == 2 || player->GetInvoke() == 4)
+    {
+        player->UpgradeGalakrond();
+    }
```
Comment (FYLSunghwan) [inline Sources/Rosetta/Tasks/SimpleTasks/InvokeTask.cpp]: 기원해서 Upgrade되는것도 테스트가 필요하지 않을까요?

Comment (utilForever) [reply Sources/Rosetta/Tasks/SimpleTasks/InvokeTask.cpp]: 'Disciple of Galakrond' 카드에서 테스트하고 있습니다.

File: Sources/Rosetta/Cards/Cards.cpp [hunk 2]
```diff
@@ -56,6 +57,17 @@ Cards::Cards()
...
             m_allWildCards.emplace_back(card);
         }
+
+        if (card->id == "DAL_613" ||  // DAL_613: Faceless Lackey
+            card->id == "DAL_614" ||  // DAL_614: Kobold Lackey
+            card->id == "DAL_615" ||  // DAL_615: Witchy Lackey
+            card->id == "DAL_739" ||  // DAL_739: Goblin Lackey
+            card->id == "DAL_741" ||  // DAL_741: Ethereal Lackey
+            card->id == "ULD_616" ||  // ULD_616: Titanic Lackey
+            card->id == "DRG_052")    // DRG_052: Draconic Lackey
```
[the remark this answers is not in the document]
Comment (utilForever) [reply Sources/Rosetta/Cards/Cards.cpp]: '졸개'를 확인할 수 있는 GameTag가 있었다면 그렇게 했을텐데, 현재 cards.json에서 졸개임을 확인할 방법이 없기 때문에 Card보다는 Cards에 추가해 확인하기 쉽게 했습니다. 의견 보시고 답변주세요.

File: Sources/Rosetta/Models/Minion.cpp [hunk 3]
```diff
@@ -30,6 +31,13 @@ void Minion::SetLastBoardPos(int value)
     SetGameTag(GameTag::TAG_LAST_KNOWN_COST_IN_HAND, value);
 }
 
+bool Minion::IsLackey() const
+{
+    auto lackeys = Cards::GetLackeys();
+
+    return std::find(lackeys.begin(), lackeys.end(), card) != lackeys.end();
+}
+
```
[the remark this answers is not in the document]
Comment (utilForever) [reply Sources/Rosetta/Models/Minion.cpp]: 위에 답변 사항 확인해주세요.

File: Sources/Rosetta/Cards/Cards.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (FYLSunghwan) [reply Sources/Rosetta/Cards/Cards.cpp]: 그 `Card::IsGalakrond()`처럼 똑같이 저 코드를  
```
bool Card::IsLackey() const
 {
    if (card->id == "DAL_613" ||  // DAL_613: Faceless Lackey
        card->id == "DAL_614" ||  // DAL_614: Kobold Lackey
        card->id == "DAL_615" ||  // DAL_615: Witchy Lackey
        card->id == "DAL_739" ||  // DAL_739: Goblin Lackey
        card->id == "DAL_741" ||  // DAL_741: Ethereal Lackey
        card->id == "ULD_616" ||  // ULD_616: Titanic Lackey
        card->id == "DRG_052")    // DRG_052: Draconic Lackey
     {
         return true;
     }

     return false;
 }
```

로 하면 나중에 `Minion.cpp`에서처럼 재사용이 가능하지 않을까 하고 생각했습니다!

File: Sources/Rosetta/Tasks/SimpleTasks/AddLackeyTask.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+{
+    auto lackeys = Cards::GetLackeys();
+
+    std::vector<Playable*> cards;
+
+    for (int i = 0; i < m_amount && !player->GetHandZone()->IsFull(); ++i)
+    {
+        const auto idx = Random::get<std::size_t>(0, lackeys.size() - 1);
+        const auto lackey = Entity::GetFromCard(
+            player, lackeys.at(idx), std::nullopt, player->GetHandZone());
```
Comment (1cekrim) [inline Sources/Rosetta/Tasks/SimpleTasks/AddLackeyTask.cpp]:
```
const auto lackey = Entity::GetFromCard(
    player, *Random::get(lackeys), std::nullopt, player->GetHandZone());
```
로 단순화 할 수 있을 것 같습니다.

File: Sources/Rosetta/Cards/Cards.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (utilForever) [reply Sources/Rosetta/Cards/Cards.cpp]: 아, 그런데 `AddLackeyTask`를 살펴보시면 왜 이렇게 했는지 아실 겁니다.

[the remark this answers is not in the document]
Comment (utilForever) [reply Sources/Rosetta/Cards/Cards.cpp]: 수정했습니다.

File: Sources/Rosetta/Tasks/SimpleTasks/AddLackeyTask.cpp (hunk 4 above)
Comment (utilForever) [reply Sources/Rosetta/Tasks/SimpleTasks/AddLackeyTask.cpp]: 수정했습니다.

**Comment by utilForever** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- Sources/Rosetta/CardSets/DragonsCardsGen.cpp  9
- Tests/UnitTests/CardSets/DalaranCardsGenTests.cpp  20
- Tests/UnitTests/CardSets/DragonsCardsGenTests.cpp  474
         

Clones removed
==============
+ Sources/Rosetta/CardSets/UldumCardsGen.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/utilForever/RosettaStone/pullRequest?prid=5095569&bid=16760792)

**Pull request merged**